File: src/chipschallenge/blockreactions/BrownButtonTo.java
package chipschallenge.blockreactions;

import chipschallenge.Block;
import chipschallenge.Buttons;
import chipschallenge.SoundPlayer.sounds;

/**
 * Moving to brown button
 */
public class BrownButtonTo extends NoSlipReaction {

    private BrownButtonTo() {
    }
    private static BrownButtonTo mInstance = null;

    public static synchronized BrownButtonTo getInstance() {
        if (mInstance == null) {
            mInstance = new BrownButtonTo();
        }
        return mInstance;
    }

    public void react(Block moving, Block standing) {
        Buttons.brownButtonDown(standing);
        sound().playSound(sounds.BUTTON);
    }

    public boolean canMove(Block moving, Block standing) {
        return true;
    }
}

File: ChipsChallenge.Shared/Blockreactions/BrownButtonTo.cs
namespace ChipsChallenge.Shared.Blockreactions
{
    public class BrownButtonTo : NoSlipReaction
    {
        private static BrownButtonTo instance;

        private BrownButtonTo()
        {
        }

        public static BrownButtonTo Instance
        {
            get
            {
                lock (typeof(BrownButtonTo))
                {
                    return instance ?? (instance = new BrownButtonTo());
                }
            }
        }

        public override void React(Block moving, Block standing)
        {
            Buttons.BrownButtonDown(standing);
            Sound().Play(Shared.Sound.Button);
        }

        public override bool canMove(Block moving, Block standing)
        {
            return true;
        }
    }
}